### DISCUSSION TITLE
Is it normal behavior or bug? Combine singular and pydantic model based body

### DISCUSSION THREAD

**Discussion by artemholovnia** [fastapi/fastapi#11046] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import FastAPI, Body, Depends
from pydantic import BaseModel, Field


app = FastAPI()


class Foo(BaseModel):
    id_: int = Field(alias="id")


def bar_dep(id_: int = Body(alias="id")) -> None:
    return


@app.post("/", dependencies=[Depends(bar_dep)])
def bar(foo: Foo) -> None:
    return
```

### Description

I have tried to combine singular Body() parameter in regular DI function and pydantic mode based body in main function, that was registered as resolver for route. After doing request
`curl -L 'http://localhost:8000' -H 'Content-Type: application/json' -d '{"id": 1}'`
i get with next response body:
```
{
    "detail": [
        {
            "loc": [
                "body",
                "foo"
            ],
            "msg": "field required",
            "type": "value_error.missing"
        }
    ]
}
```
Is It normal behavior? Why i can't access to concrete field in request in extra dependencies for some extra stuff?
Look likes that body was cleared after accessing first time and does not propagate to next part when processing data to pydantic model...
Thanks for explanation!
p.s. In example i have used a simple case thats not a final code but its a case that allow me to do something advanced.

### Operating System

macOS

### FastAPI Version

0.109.0

### Pydantic Version

1.10.10

### Python Version

Python 3.11.6

**Comment by YuriiMotov** [comment, marked as answer]
When you combine Pydantic model with single Body parameters or with another Pydantic model, FastAPI expects data for parameters defined as models to be passed inside the nested dictionary that should be accessible by key=parameter_name.

In your case the expected input payload is:

```
{
  "foo": {
    "id": 0
  },
  "id": 0
}
```
So, all single parameters are on the first level, but parameters defined as model are inside the "foo" field.
